pkg/virtctl/create/vm/vm.go

```diff
@@ -318,8 +229,21 @@ func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
 		snapshot, params.Supported(dataVolumeSourceSnapshot{}),
 	))
 
+	cmd.Flags().StringVar(&c.user, UserFlag, c.user, "Specify the user in the cloud-init user data that is added to the VM.")
+	cmd.Flags().BoolVar(&c.genPassword, GenPasswordFlag, c.genPassword, "Specify if a password should be generated in the cloud-init user data that is added to the VM.")
+	cmd.Flags().StringVar(&c.passwordFile, PasswordFileFlag, c.passwordFile, "Specify a file to read the password from for the cloud-init user data that is added to the VM.")
+	cmd.Flags().StringSliceVar(&c.sshKeys, SSHKeyFlag, c.sshKeys, "Specify one or more SSH authorized keys in the cloud-init user data that is added to the VM.")
+	cmd.Flags().BoolVar(&c.gaManageSSH, GAManageSSHFlag, c.gaManageSSH, "Specify if the qemu-guest-agent should be able to manage SSH in the cloud-init user data that is added to the VM.")
+	cmd.MarkFlagsMutuallyExclusive(GenPasswordFlag, PasswordFileFlag)
+
+	cmd.Flags().StringVar(&c.cloudInitSourceType, CloudInitSourceTypeFlag, c.cloudInitSourceType, "Specify the cloud-init data source type. Allowed values are noCloud or configDrive.")
```

Comment:
Are we aware of any reason why the user might want to prefer configDrive over ncloud, or would it simplify the code if only nocloud would work?

Reply:
It was a TODO before to add support for config drives. IMO there is only slight overhead for supporting both so I'd keep it.